docs/recipes/virtualenv.rst

```diff
@@ -0,0 +1,65 @@
+Setting Up a Python Virtualenv for SCons
+----------------------------------------
+
+It is often useful to set up a virtualenv when working with a project
+that uses SCons to build.  A virtualenv is a way to create an
+isolated execution environment - you can install whatever you need
+there, and change versions as needed, without affecting anything
+else on your system that uses Python.  Here is an example session
+for setting one up:
+
+.. code:: shell
+
+   $ cd Work
+   $ python -m venv myvenv
+   $ cd myvenv
+   $ source bin/activate
+   (myvenv) $ python -m pip list --outdated
+   Package    Version Latest Type
+   ---------- ------- ------ -----
+   pip        19.3.1  21.0.1 wheel
+   setuptools 41.6.0  53.0.0 wheel
+   WARNING: You are using pip version 19.3.1; however, version 21.0.1 is available.
+   You should consider upgrading via the 'pip install --upgrade pip' command.
+   (myvenv) $ python -m pip install --upgrade pip setuptools scons
```

Comment:
But not here, and maybe include a comment about why.. though `pip install -U pip` works on non-windows, it fails on windows.

Reply:
Took a moment to parse this.  Anyway, I have the equivalent Windows log from the same time, just wasn't sure if it needed to be added to the recipe.
```
C:\Users\mats>cd Work

C:\Users\mats\Work>py -m venv myvenv

C:\Users\mats\Work>cd myvenv

C:\Users\mats\Work\myvenv>.\Scripts\activate

(myvenv) C:\Users\mats\Work\myvenv>python -m pip list --outdated
Package    Version Latest Type
---------- ------- ------ -----
pip        20.2.3  21.0.1 wheel
setuptools 49.2.1  53.0.0 wheel
WARNING: You are using pip version 20.2.3; however, version 21.0.1 is available.You should consider upgrading via the 'C:\Users\mats\Work\myvenv\Scripts\python.exe -m pip install --upgrade pip' command.

(myvenv) C:\Users\mats\Work\myvenv>python -m pip install --upgrade pip setuptools scons
Collecting pip
  Using cached pip-21.0.1-py3-none-any.whl (1.5 MB)
Collecting setuptools
  Using cached setuptools-53.0.0-py3-none-any.whl (784 kB)
Collecting scons
  Using cached SCons-4.1.0.post1-py3-none-any.whl (4.1 MB)
Installing collected packages: pip, setuptools, scons
  Attempting uninstall: pip
    Found existing installation: pip 20.2.3
    Uninstalling pip-20.2.3:
      Successfully uninstalled pip-20.2.3
  Attempting uninstall: setuptools
    Found existing installation: setuptools 49.2.1
    Uninstalling setuptools-49.2.1:
      Successfully uninstalled setuptools-49.2.1
Successfully installed pip-21.0.1 scons-4.1.0.post1 setuptools-53.0.0

(myvenv) C:\Users\mats\Work\myvenv>scons --version
SCons by Steven Knight et al.:
        SCons: v4.1.0.post1.dc58c175da659d6c0bb3e049ba56fb42e77546cd, 2021-01-20 04:32:28, by bdbaddog on ProDog2020
        SCons path: ['c:\\users\\mats\\work\\myvenv\\lib\\site-packages\\SCons']
Copyright (c) 2001 - 2021 The SCons Foundation

(myvenv) C:\Users\mats\Work\myvenv>
```